React/React hooks.md

```diff
@@ -13,19 +74,68 @@ export const MyComponent: FC = ({ numberProp, stringProps }) => {
 }
 ```
 
-For non-primitive values like objects, arrays and functions, React will do a reference comparison, instead of comparing them with their previous values.
+For non-primitive values like objects, arrays and functions, React will do a reference comparison using [Object.is()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is). This means that we must provide the same reference of `objectProp`, `arrayProp`, `functionProp` to `Child` on every re-render if we don't want to trigger `useEffect` every time. This can be achieved by extracting them outside component function or wrapping them in `useMemo` or `useCallback`.  
```

Comment:
```suggestion
For non-primitive values like objects, arrays and functions, React will do a reference comparison using [Object.is()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is). This means that we must provide the same reference of `objectProp`, `arrayProp`, `functionProp` to `Child` on every re-render if we don't want to trigger `useEffect` every time. This can be achieved by extracting them outside of the component or wrapping them in `useMemo` or `useCallback`.  
```